driver-core/src/main/com/mongodb/internal/session/ServerSessionPool.java

```diff
@@ -56,6 +56,7 @@
     private volatile boolean closing;
     private volatile boolean closed;
     private final List<BsonDocument> closedSessionIdentifiers = new ArrayList<BsonDocument>();
+    private final Object closeLock = new Object();
```

Comment:
Do you need a separate lock object?  You can synchronize on `this` instead.

Reply:
Done.